pkg/controllers/managementuser/rbac/cluster_handler.go

```diff
@@ -67,47 +67,51 @@ func (h *clusterHandler) sync(key string, obj *v3.Cluster) (runtime.Object, erro
 
 func (h *clusterHandler) doSync(cluster *v3.Cluster) error {
 	_, err := v32.ClusterConditionGlobalAdminsSynced.DoUntilTrue(cluster, func() (runtime.Object, error) {
-		// Sync both admin types
-		for _, roleName := range []string{rbac.GlobalAdmin, rbac.GlobalRestrictedAdmin} {
-			// Do not sync restricted-admin to the local cluster as 'cluster-admin'
-			if cluster.Name == "local" && roleName == rbac.GlobalRestrictedAdmin {
-				continue
-			}
-			grbs, err := h.grbIndexer.ByIndex(grbByRoleIndex, roleName)
-			if err != nil {
+		// For restricted admins, re-enqueue the GRBs to trigger the creation of any RBAC resources on new cluster creation.
+		grbs, err := h.grbIndexer.ByIndex(grbByRoleIndex, rbac.GlobalRestrictedAdmin)
+		if err != nil {
+			return nil, err
+		}
+		for _, x := range grbs {
+			grb, _ := x.(*v3.GlobalRoleBinding)
+			h.grbController.Enqueue("", grb.Name)
+		}
+
+		grbs, err = h.grbIndexer.ByIndex(grbByRoleIndex, rbac.GlobalAdmin)
+		if err != nil {
+			return nil, err
```

Comment:
Nit: If you could wrap the error here indicating that this failed for the admin index retrieval that would be good.

Reply:
done